examples/celllistmap.jl

```diff
@@ -0,0 +1,249 @@
+# out of the box. In the future, Agents.jl may also switch the `pos` type to a static vector.
+
+# We `import` the `CellListMap` package because
+# it exports some functions with conflicting names with `Agents`.
+
+# 
+# ## Model properties.
+# 
+# The `clmap_system` field will store the data required for `CellListMap`, and again we 
+# use a parametric type to guarantee the concrectness of the types, aoviding type
+# instabilities.  
+#
+Base.@kwdef struct Properties{CL<:CellListMap.PeriodicSystem}
+    dt::Float64 = 0.01
+    number_of_particles::Int64 = 0
+    clmap_system::CL # CellListMap system
+end
+
+# Two auxiliary arrays will be created on model initialization, to be passed to
+# the `CellListMap.PeriodicSystem` data structure:
+#
+# 1. `positions`: `CellListMap` requires a vector of (preferentially) static vectors as the positions
+#    of the particles. To avoid creating this array on every call, a buffer to
+#    which the `agent.pos` positions will be copied is stored in this data structure.
+# 2. `forces`: In this example, the property to be computed using `CellListMap.jl` is 
+#    the forces between particles, which are stored here in a `Vector{<:SVector}`, of
+#    the same type as the positions. These forces will be updated by the `CellListMap.map_pairwise!`
+#    function.
+#
+# Additionally, the computation with `CellListMap.jl` requires the definition of a `cutoff`,
+# which will be twice the maximum interacting radii of the particles, and the geometry of the
+# the system, given by the `sides` of the periodic box. 
+#
+# Just for clarity, we define an alias for the function that fetches the forces from 
+# the `CellListMap.PeriodicSystem` structure:
+const get_forces = CellListMap.get_output
+
+#
+# ## Model initialization
+# We create the model with a keyword-accepting function as is recommended in Agents.jl.
+# The keywords here control number of particles and sizes.
+function initialize_model(;
+    number_of_particles=10_000,
+    sides=SVector(500.0, 500.0),
+    dt=0.001,
+    max_radius=10.0,
+    parallel=true
+)
+    ## initial random coordinates
+    coordinates = [sides .* rand(SVector{2,Float64}) for _ in 1:number_of_particles]
+
+    ## We will use CellListMap to compute forces, with similar structure as the coordinates
+    forces = similar(coordinates)
+
+    ## Space and agents
+    space2d = ContinuousSpace(Tuple(sides); periodic=true)
+
+    ## default maximum radius is 10.0 thus cutoff is 20.0
+    cutoff = 2 * max_radius
+
+    ## Initialize CellListMap periodic system
+    clmap_system = CellListMap.PeriodicSystem(
+        coordinates=coordinates,
+        sides=sides,
+        cutoff=cutoff,
+        output=forces,
+        parallel=parallel,
+    )
+
+    ## define the model properties
+    properties = Properties(
+        dt=dt,
+        number_of_particles=number_of_particles,
+        clmap_system=clmap_system,
+    )
```

Comment:
There is no reason to make the additional struct `Properties` here. Instead, you should use a standard named tuple. These three "properties" are constants, they do not change during the simulation, hence a standard named tuple is enough.

Reply:
(Named Tuples are also strictly typed)